### PR TITLE
MCSH debug and example

### PR DESCRIPTION
**Pull request by ray38**

### CODE REVIEW

File: amptorch/descriptor/MCSH/__init__.py [hunk 1]
```diff
@@ -185,26 +186,35 @@ def calculate_fingerprints(self, atoms, element, log=None, calculate_derivatives
...
-        if "prime_threshold" in self.params_set:
-            threshold = self.params_set["prime_threshold"]
-            super_threshold_indices = np.abs(fp_prime) < threshold
-            print("threshhold: {} \tnum points set to zero:{} \t outof: {}".format(threshold, np.sum(super_threshold_indices), fp_prime.shape[0] * fp_prime.shape[1]))
-            fp_prime[super_threshold_indices] = 0.0
+        # print(fp)
+        # print(fp_prime)
+
+        # if "prime_threshold" in self.params_set:
+        #     threshold = self.params_set["prime_threshold"]
```
Comment (mshuaibii) [inline amptorch/descriptor/MCSH/__init__.py:198]: Can you remove all the commented out logic in the MCSH files.

File: amptorch/descriptor/base_descriptor.py [hunk 2]
```diff
@@ -172,6 +174,11 @@ def _compute_fingerprints(
                         num_desc_dict[element] = size_info[2]
                         fp_dict[element] = fps
 
+                        # print(len(fps[0]))
```
Comment (mshuaibii) [inline amptorch/descriptor/base_descriptor.py:177]: Can you remove them here as well

File: amptorch/trainer.py [hunk 3]
```diff
@@ -76,10 +77,20 @@ def load_dataset(self):
...
+        fp_scheme = self.config["dataset"].get("fp_scheme", "gaussian").lower()
+        
+        if fp_scheme == "gaussian":
+            fp_params = self.config["dataset"]["fp_params"]
+            self.descriptor = Gaussian(Gs=fp_params, elements=self.elements)
+        elif fp_scheme == "mcsh":
+            fp_params = self.config["dataset"]["fp_params"]
+            self.descriptor = AtomisticMCSH(MCSHs=fp_params, elements=self.elements)
+        else:
+            raise NotImplementedError
```
Comment (mshuaibii) [inline amptorch/trainer.py:90]: Let's move this logic to the `AtomsDataset` instead. The `descriptor` argument for `AtomsDataset` can now be a string  in which this above logic is applied. We want to keep the trainer as clean as possible.

File: examples/mcsh_example.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                        "4": {"groups": [1,2,3,4], "sigmas": sigmas},
+                        "5": {"groups": [1,2,3,4,5], "sigmas": sigmas},
+                        "6": {"groups": [1,2,3,4,5,6,7], "sigmas": sigmas},
+                        #"7": {"groups": [1,2,3,4,5,6,7,8], "sigmas": sigmas},
+                        #"8": {"groups": [1,2,3,4,5,6,7,8,9,10], "sigmas": sigmas},
+                        #"9": {"groups": [1,2,3,4,5,6,7,8,9,10,11,12], "sigmas": sigmas}
+                  },
+            "atom_gaussians": {
+            "C": "./MCSH_potential/C_coredensity_5.g",
+			"H": "./MCSH_potential/H_totaldensity_3.g",
```
Comment (mshuaibii) [inline examples/mcsh_example.py:39]: Are these files required for MCSH? If so, we should probably include them for MCSH use. If they're above >10MB though we can make it a separate download instead though.

**Review by mshuaibii — CHANGES_REQUESTED** [review]
Looks good - minor changes/clarifications.

File: amptorch/descriptor/MCSH/__init__.py (hunk 1 above)
Comment (ray38) [reply amptorch/descriptor/MCSH/__init__.py:198]: I'll clean it up as much as I could,
though I would probably need to leave some there for debug purpose for now. once I'm convinced everything works as it should, I'll clean everything up

File: amptorch/descriptor/base_descriptor.py (hunk 2 above)
Comment (ray38) [reply amptorch/descriptor/base_descriptor.py:177]: sure

File: amptorch/trainer.py (hunk 3 above)
Comment (ray38) [reply amptorch/trainer.py:90]: OK, I'll make the change

File: examples/mcsh_example.py (hunk 4 above)
Comment (ray38) [reply examples/mcsh_example.py:39]: yes, you can think them as parameter files for different atom types, I can include some example ones in the examples directory

File: amptorch/trainer.py (hunk 3 above)
Comment (ray38) [reply amptorch/trainer.py:90]: Actually, I think we need to clarify this a bit, 
if we want to move this logic to be in AtomsDataset, we need to add descriptor_type, descriptor_config and elements as extra arguments to that object, and somehow I think this is not as clean as just taking an descriptor object
also the trainer won't have access to the descriptor (tho we might not need it to anyway)

File: amptorch/descriptor/MCSH/__init__.py (hunk 1 above)
Comment (mshuaibii) [reply amptorch/descriptor/MCSH/__init__.py:192]: Yup no worries if intentional, thought it was missed.

File: amptorch/trainer.py (hunk 3 above)
Comment (ray38) [reply amptorch/trainer.py:90]: Or, actually, maybe we can put this part into util, which I think it's the cleanest way

Comment (mshuaibii) [reply amptorch/trainer.py:90]: Maybe just make a method in dataset.py instead, like I did with the collater - so we centralize methods/classes rather than putting it in utils.
